docs/design/permissive-mode.md

```diff
@@ -0,0 +1,353 @@
+
+JSON Schema: [https://cyclonedx.org/docs/1.6/json](https://cyclonedx.org/docs/1.6/json)
+
+Validation can be done using the CycloneDX [CLI](https://github.com/CycloneDX/cyclonedx-cli)
+tool.
+
+```bash
+cyclonedx validate --input-file bom.json
+```
+
+#### Approach 1 (preferred): Top-level annotation with component references
+
+This approach involves adding top-level annotations that describe permissive mode
+violations and referencing the affected components by their IDs.
+
+Example:
+
+```json
+{
+  "bomFormat": "CycloneDX",
+  "annotations": [
+    {
+      "subjects": ["123abc", "123xyz"],
+      "annotator": {
+        "organization": {
+          "name": "red hat"
+        }
+      },
+      "timestamp": "2025-01-15T10:30:00Z",
+      "text": "hermeto:permissive-mode:violation:gomod:vendor-directory-changed-after-vendoring"
+    }
+  ],
+  "components": [
+    {
+      "bom-ref": "123abc",
+      "name": "package",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        }
+      ],
+      "purl": "pkg:golang/github.com/example/package@v1.2.3",
+      "type": "library",
+      "version": "v1.2.3"
+    },
+    {
+      "bom-ref": "123xyz",
+      "name": "package",
+
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        }
+      ],
+      "purl": "pkg:golang/github.com/another/package@v2.1.0",
+      "type": "library",
+      "version": "v2.1.0"
+    }
+  ],
+  "specVersion": "1.6",
+  "version": 1,
+  "metadata": {
+    "tools": [
+      {
+        "vendor": "red hat",
+        "name": "hermeto"
+      }
+    ]
+  }
+}
+```
+
+#### Approach 2 (alternative): Properties within the components
+
+This approach involves adding properties that describe the violation directly
+to each affected component.
+
+Example:
+
+```json
+{
+  "bomFormat": "CycloneDX",
+  "components": [
+    {
+      "name": "github.com/gin-gonic/gin",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        },
+        {
+          "name": "hermeto:permissive-mode:violation",
+          "value": "gomod:vendor-directory-changed-after-vendoring"
+        }
+      ],
+      "purl": "pkg:golang/github.com/gin-gonic/gin@v1.9.1",
+      "type": "library",
+      "version": "v1.9.1"
+    },
+    {
+      "name": "github.com/gorilla/mux",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        },
+        {
+          "name": "hermeto:permissive-mode:violation",
+          "value": "gomod:vendor-directory-changed-after-vendoring"
+        }
+      ],
+      "purl": "pkg:golang/github.com/gorilla/mux@v1.8.0",
+      "type": "library",
+      "version": "v1.8.0"
+    }
+  ],
+  "specVersion": "1.6",
+  "version": 1,
+  "metadata": {
+    "tools": [
+      {
+        "vendor": "red hat",
+        "name": "hermeto"
+      }
+    ]
+  }
+}
+```
+
+### SPDX
```

Comment:
The very important bit of information that is missing is that we are using SPDX v2.3 and are unlikely to switch to v3.0 in the near future. I think it is necessary to add a clear statement that by SPDX in this document you mean v2.3 unless stated otherwise.